src/selectors/entities/schemes.js

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2018-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+// @flow
+
+import {getAllChannels} from 'selectors/entities/channels';
+import {getTeams} from 'selectors/entities/teams';
+import {ScopeTypes} from 'constants/schemes';
+import type {GlobalState} from 'types/store';
+import type {Scheme} from 'types/schemes';
+import type {Channel} from 'types/channels';
+import type {Team} from 'types/teams';
+
+export function getSchemes(state: GlobalState): { [string]: Scheme } {
+    return state.entities.schemes.schemes;
+}
+
+export function getScheme(state: GlobalState, id: string): Scheme {
+    const schemes = getSchemes(state);
+    return schemes[id];
+}
+
+export function getSchemeChannels(state: GlobalState, schemeId: string): Array<Channel> {
+    const {scope} = getScheme(state, schemeId);
+
+    if (scope === ScopeTypes.TEAM) {
+        throw new Error(`Not implemented: scheme '${schemeId}' is team-scope but 'getSchemeChannels' only accepts channel-scoped schemes.`);
+    }
+
+    const schemeChannels: Array<Channel> = [];
+
+    Object.entries(getAllChannels(state)).forEach((item: [string, Channel]) => {
+        const [, channel: Channel] = item;
+        if (channel.scheme_id === schemeId) {
+            schemeChannels.push(channel);
+        }
+    });
+
+    return schemeChannels;
```

Comment:
Also any thoughts about returning an array vs object here?

Reply:
0/5, I usually lean towards an array if it's likely I'm to need to order the results and an object otherwise but I have no strong opinion